tests/server/tso/tso_test.go

```diff
@@ -93,3 +94,55 @@ func requestLocalTSOs(re *require.Assertions, cluster *tests.TestCluster, dcLoca
 	}
 	return tsMap
 }
+
+func TestDisableLocalTSOAfterEnabling(t *testing.T) {
+	re := require.New(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	dcLocationConfig := map[string]string{
+		"pd1": "dc-1",
+		"pd2": "dc-2",
+		"pd3": "dc-3",
+	}
+	dcLocationNum := len(dcLocationConfig)
+	cluster, err := tests.NewTestCluster(ctx, dcLocationNum, func(conf *config.Config, serverName string) {
+		conf.EnableLocalTSO = true
+		conf.Labels[config.ZoneLabel] = dcLocationConfig[serverName]
+	})
+	defer cluster.Destroy()
+	re.NoError(err)
+	re.NoError(cluster.RunInitialServers())
+
+	cluster.WaitAllLeaders(re, dcLocationConfig)
+	requestLocalTSOs(re, cluster, dcLocationConfig)
+	re.NoError(failpoint.Enable("github.com/tikv/pd/server/tso/systemTimeSlow", `return(true)`))
+
+	// Reboot the cluster.
+	re.NoError(cluster.StopAll())
+	for _, server := range cluster.GetServers() {
+		server.SetEnableLocalTSO(false)
+		server.GetTSOAllocatorManager().SetEnableLocalTSO(false)
+	}
+
+	re.NoError(cluster.RunInitialServers())
+	cluster.WaitLeader()
+
+	// Re-request the global TSOs.
+	leaderServer := cluster.GetServer(cluster.GetLeader())
+	grpcPDClient := testutil.MustNewGrpcClient(re, leaderServer.GetAddr())
+	clusterID := leaderServer.GetClusterID()
+	req := &pdpb.TsoRequest{
+		Header:     testutil.NewRequestHeader(clusterID),
+		Count:      1,
+		DcLocation: tso.GlobalDCLocation,
+	}
+
+	ctx = grpcutil.BuildForwardContext(ctx, leaderServer.GetAddr())
+	tsoClient, err := grpcPDClient.Tso(ctx)
+	re.NoError(err)
+	defer tsoClient.CloseSend()
+	re.NoError(tsoClient.Send(req))
+	resp, err := tsoClient.Recv()
+	re.NoError(err)
+	re.NotNil(checkAndReturnTimestampResponse(re, req, resp))
```

Comment:
In addition to requesting a Global TSO, I think we can also call `GetClusterDCLocationsFromEtcd` to check if the number of existing DCs is as expected.